src/overlays/actors/ovl_En_Lift_Nuts/z_en_lift_nuts.c

```diff
@@ -28,111 +62,929 @@ const ActorInit En_Lift_Nuts_InitVars = {
+        Player* player = GET_PLAYER(play);
+
+        player->stateFlags1 |= 0x20;
+        func_80AE9AC4(this, 1);
+        func_80AE9B4C(1, 3);
+        func_80AEA0B4(this);
+    } else {
+        func_80AE9F28(this);
+    }
+}
+
+void EnLiftNuts_Destroy(Actor* thisx, PlayState* play) {
+    EnLiftNuts* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+    func_80AE9A80(this, play);
+}
+
+void func_80AE9F28(EnLiftNuts* this) {
+    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 18);
+    this->actionFunc = func_80AE9F70;
+    this->unk_354 = 0;
+}
+
+void func_80AE9F70(EnLiftNuts* this, PlayState* play) {
+    if (this->unk_354 < 40) {
+        this->unk_354++;
+    } else if (this->actor.xzDistToPlayer < 100.0f) {
+        func_80AEA0B4(this);
+    }
+}
+
+void func_80AE9FC8(EnLiftNuts* this) {
+    if (this->actionFunc == func_80AEA1A0) {
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 17);
+    } else {
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 3);
+    }
+
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_AKINDONUTS_HIDE);
+    this->actionFunc = func_80AEA044;
+}
+
+void func_80AEA044(EnLiftNuts* this, PlayState* play) {
+    if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+        func_80AE9F28(this);
+    } else if (Animation_OnFrame(&this->skelAnime, 5.0f)) {
+        func_80AEB9E0(this, play);
+    }
+}
+
+void func_80AEA0B4(EnLiftNuts* this) {
+    if (func_80AE9B4C(0, 1)) {
+        this->actionFunc = func_80AEA1A0;
+    } else {
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 2);
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_AKINDONUTS_HIDE);
+        this->actionFunc = func_80AEA128;
+    }
+}
+
+void func_80AEA128(EnLiftNuts* this, PlayState* play) {
+    if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 15);
+        this->actionFunc = func_80AEA1A0;
+    } else if (Animation_OnFrame(&this->skelAnime, 8.0f)) {
+        func_80AEB9E0(this, play);
+    }
+}
+
+void func_80AEA1A0(EnLiftNuts* this, PlayState* play) {
+    s32 pad;
+    OSTime time;
+
+    if ((func_80AE9B4C(0, 3) || func_80AE9B4C(0, 1)) && this->unk_34E == 1) {
+        this->actor.flags |= 0x10000;
+    } else if (this->actor.xzDistToPlayer > 120.0f) {
+        func_80AE9FC8(this);
+    }
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        if (gSaveContext.save.playerForm == PLAYER_FORM_DEKU) {
+            if (func_80AE9B4C(0, 0)) {
+                switch (CURRENT_DAY) {
+                    case 1:
+                        if (gSaveContext.save.time >= 0xFAAB || gSaveContext.save.time <= 0x4000) {
+                            Message_StartTextbox(play, 0x27F7, &this->actor);
+                            this->textId = 0x27F7;
+                        } else if (gSaveContext.save.weekEventReg[0xE] & 0x10) {
```

Comment:
```suggestion
                        } else if (gSaveContext.save.weekEventReg[14] & 0x10) {
```
Note this applies elsewhere.